client/components/LearnerViewDirective.js

```diff
@@ -71,6 +50,27 @@ tie.directive('learnerView', [function() {
               </div>
             </div>
           </div>
+          <div class="tie-question-ui">
+            <div class="tie-question-window">
+              <h3>Exercise {{exerciseNumber}}: {{title}}</h3>
```

Comment:
The internal properties of exercises _are_ separate from the view presented to the user --that is the point of the JS data files.

However, the fact that a particular exercise is "Exercise 1" is not an internal property of the exercise. It's a property of where it is in the set-of-exercises, and the latter is where this information should be maintained. Currently, that is done in the application logic, but I think that it would be better for this to be more data driven and I think that in the future we should introduce a set-of-questions data object as well. I will look into this.

With regards to how to display "Exercise 1" (as opposed to, say, "Question 1"?), that's at the level of either a set-of-exercises, or even higher (e.g. if we use the same display for all sets of exercises within the application). I'm not sure whether, by UI judgement call, you mean a customization at the same level as the color of the background of the page (for example), but if that's the case I see no reason why it should be treated any differently for the present (i.e. just included as part of the code).